libyul/Object.h

```diff
@@ -125,6 +126,13 @@ struct Object: public ObjectNode
 
 	/// @returns the name of the special metadata data object.
 	static std::string metadataName() { return ".metadata"; }
+
+	/// Recursively searches for an Object at @param _qualifiedPath within @param _object.
+	/// @returns a shared_ptr to the Object or a nullptr if it was not found.
+	static std::shared_ptr<Object> objectAt(
+		std::shared_ptr<Object> const& _object,
+		std::string const& _qualifiedPath
```

Comment:
I'd call it `qualifiedName` just `pathToSubObject()`:
```suggestion
		std::string const& _qualifiedName
```

But you have a point that it's a path. It's just that in this file by path we seem to mean specifically a vector with indexes so I'd stick to that for consistent terminology.